src/Http/Routing/src/RouteEndpointDataSource.cs

```diff
@@ -69,10 +157,34 @@ public RouteEndpointDataSource(IServiceProvider applicationServices, bool throwO
             HttpMethods = httpMethods,
             RouteAttributes = routeAttributes,
             Conventions = conventions,
-            FinallyConventions = finallyConventions
+            FinallyConventions = finallyConventions,
+            InferMetadataFunc = InferHandlerMetadata,
+            CreateHandlerRequestDelegateFunc = CreateHandlerRequestDelegate
         });
 
         return new RouteHandlerBuilder(conventions, finallyConventions);
+
+        [UnconditionalSuppressMessage("Trimmer", "IL2026",
+            Justification = "We surface a RequireUnreferencedCode in the call to the Map methods adding route handlers to this EndpointDataSource. Analysis is unable to infer this. " +
+            "Map methods that configure a RequestDelegate don't use trimmer unsafe features.")]
+        [UnconditionalSuppressMessage("AOT", "IL3050",
+            Justification = "We surface a RequiresDynamicCode in the call to the Map methods adding route handlers this EndpointDataSource. Analysis is unable to infer this. " +
+            "Map methods that configure a RequestDelegate don't use AOT unsafe features.")]
+        static RequestDelegateMetadataResult InferHandlerMetadata(MethodInfo methodInfo, RequestDelegateFactoryOptions? options = null)
+        {
+            return RequestDelegateFactory.InferMetadata(methodInfo, options);
+        }
+
+        [UnconditionalSuppressMessage("Trimmer", "IL2026",
+            Justification = "We surface a RequireUnreferencedCode in the call to the Map methods adding route handlers to this EndpointDataSource. Analysis is unable to infer this. " +
+            "Map methods that configure a RequestDelegate don't use trimmer unsafe features.")]
+        [UnconditionalSuppressMessage("AOT", "IL3050",
+            Justification = "We surface a RequiresDynamicCode in the call to the Map methods adding route handlers this EndpointDataSource. Analysis is unable to infer this. " +
+            "Map methods that configure a RequestDelegate don't use AOT unsafe features.")]
+        static RequestDelegate CreateHandlerRequestDelegate(Delegate handler, RequestDelegateFactoryOptions options, RequestDelegateMetadataResult? metadataResult)
+        {
+            return RequestDelegateFactory.Create(handler, options, metadataResult).RequestDelegate;
+        }
```

Comment:
Why are these still in this file? Can you move these to Map* calls?

Reply:
We should be able to remove the `[UnconditionalSuppressMessage]` attributes here and also remove `[RequireUnreferenced/DynamicCode]` attributes from `EndpointRouteBuilderExtensions`, right? Wouldn't the warnings from our `[RequireUnreferenced/DynamicCode]` attributes in RDF then bubble up through the Map* calls? I thought that was the reason we decided to store the funcs in the `RouteEntry`.